tests/server-integration/Tag/TagTest.hs

```diff
@@ -0,0 +1,86 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+module Tag.TagTest
+  ( unit_tag_entry
+  , unit_delete_tag
+  ) where
+
+import Data.Aeson
+import Data.Aeson.QQ.Simple (aesonQQ)
+import Data.Functor ((<&>))
+import Data.Text (Text)
+import Network.HTTP.Req
+import Test.Tasty.HUnit
+import Utils
+
+addOrRemoveTag
+  ::
+   ( HttpMethod method
+   , HttpBodyAllowed (AllowsBody method) 'NoBody
+   )
+   => method -> Text -> Text -> IO (JsonResponse Value)
+addOrRemoveTag method path tag = runReq defaultHttpConfig do
+  req
+    method
+    (baseUrl /: "tag")
+    NoReqBody
+    (jsonResponse @Value)
+    ( mconcat
+        [ header "token" "root"
+        , "path" =: path
+        , "tag" =: tag
+        , port 8081
+        ]
+    )
+
+unit_tag_entry :: IO ()
+unit_tag_entry = cofferTest do
+  createEntry "tagged"
+
+  response <- addOrRemoveTag POST "tagged" "tag"
+
+  value <- getValueContents (responseBody response) <&> scrubDates
+  value @?= expected
+  where
+    expected =
+      [aesonQQ|
+        {
+          "ePath": {
+              "unEntryPath": [
+                  "tagged"
+              ]
+          },
+          "eFields": {},
+          "eDateModified": "",
+          "eMasterField": null,
+          "eTags": [
+              "tag"
+          ]
+        }
+      |]
+
+unit_delete_tag :: IO ()
+unit_delete_tag = cofferTest do
+  unit_tag_entry
+
+  response <- addOrRemoveTag DELETE "tagged" "tag"
+
+  value <- getValueContents (responseBody response) <&> scrubDates
+  value @?= expected
+  where
+    expected =
+      [aesonQQ|
+        {
+          "ePath": {
+              "unEntryPath": [
+                  "tagged"
+              ]
+          },
+          "eFields": {},
+          "eDateModified": "",
+          "eMasterField": null,
+          "eTags": []
+        }
+      |]
```

Comment:
Can we please also add tests for failure scenarios, and check that the correct status code is returned + the response body has the expected content?

Reply:
I left the comment on this test module, but it applies to all modules.

Reply:
Done